packages/liquidation-sdk-viem/src/preLiquidation/logGetters.ts

```diff
@@ -0,0 +1,148 @@
+import { type Address, ChainUtils, type MarketId } from "@morpho-org/blue-sdk";
+import { addresses } from "@morpho-org/blue-sdk";
+import type { Account, Chain, Client, Transport } from "viem";
+import { getBlockNumber, getLogs } from "viem/actions";
+import { preLiquidationFactoryConfigs } from "../addresses";
+import type { PreLiquidation } from "./types";
+
+export async function preLiquidationLogs<chain extends Chain = Chain>(
+  client: Client<Transport, chain, Account>,
+): Promise<PreLiquidation[]> {
+  const chainId = ChainUtils.parseSupportedChainId(client.chain.id);
+
+  try {
+    const head = await getBlockNumber(client);
+
+    const logs = await getLogs(client, {
+      address: preLiquidationFactoryConfigs[chainId].address,
+      fromBlock: preLiquidationFactoryConfigs[chainId].deploymentBlock,
+      toBlock: BigInt(head),
+      event: {
```

Comment:
it would be worth moving the definition to the `abi.ts` file to make this file more readable